backend/src/main/scala/bloop/tracing/BraveTracer.scala

```diff
@@ -26,34 +26,46 @@ final class BraveTracer private (
   }
 
   def trace[T](name: String, tags: (String, String)*)(
-      thunk: BraveTracer => T
+      thunk: BraveTracer => T,
+      verbose: Boolean = false
   ): T = {
-    val newTracer = startNewChildTracer(name, tags: _*)
-    try thunk(newTracer) // Don't catch and report errors in spans
-    catch {
-      case NonFatal(t) =>
-        newTracer.currentSpan.error(t)
-        throw t
-    } finally {
-      try newTracer.terminate()
-      catch { case NonFatal(t) => () }
+    if (!verbose || BraveTracer.verboseTrace) {
+      val newTracer = startNewChildTracer(name, tags: _*)
+      try thunk(newTracer) // Don't catch and report errors in spans
+      catch {
+        case NonFatal(t) =>
+          newTracer.currentSpan.error(t)
+          throw t
+      } finally {
+        try newTracer.terminate()
+        catch {
+          case NonFatal(t) => ()
+        }
+      }
+    } else {
+      thunk(this)
     }
   }
 
   def traceTask[T](name: String, tags: (String, String)*)(
```

Comment:
nit: should we add a `def traceVerboseTask`? It doesn't read nicely with the `thunk` lambda on the same parameter list as the `verbose` flag. We could have a `private def traceTaskInternal` that takes all the parameters, which the public `traceTask` and `traceVerboseTask` forward to

Reply:
:+1: Could be something like `private def traceTaskInternal(name: String, verbose: Boolean, tags: (String, String*))(thunk: BraveTracer => Task[T]): Task[T]`